### PR TITLE
Create version 1 test runner

### PR DESCRIPTION
**Pull request by m-dango**

### CODE REVIEW

**Review by ingydotnet — APPROVED** [review]

File: tests/all-fail/expected_results.json
Comment (ErikSchierboom) [inline tests/all-fail/expected_results.json:1]: Why did you remove this test case?

File: tests/empty-file/expected_results.json
Comment (ErikSchierboom) [inline tests/empty-file/expected_results.json:1]: Why did you remove this test case?

File: tests/all-fail/expected_results.json
Comment (ingydotnet) [reply tests/all-fail/expected_results.json:1]: Not sure. @m-dango that was from your commit e5dd386851 if I'm not mistaken.

Comment (ingydotnet) [reply tests/all-fail/expected_results.json:1]: I'll try adding them back.

File: tests/empty-file/expected_results.json
Comment (ingydotnet) [reply tests/empty-file/expected_results.json:1]: Implemented in a270fcc

File: tests/all-fail/expected_results.json
Comment (ingydotnet) [reply tests/all-fail/expected_results.json:1]: Added back and implemented in 91d5c7d

**Comment by m-dango** [comment]
@ingydotnet The TAP output looks off, it seems some of the tests indicate they're expecting `nil` rather than `false`?

File: Dockerfile [hunk 1]
```diff
@@ -1,8 +1,36 @@
...
+
+# Install packages required to run the tests:
+RUN apt-get update \
+ && apt-get install -y apt-utils \
+ && apt-get install -y \
+        curl \
+        jq \
+        make \
+        perl \
+        wget \
```
Comment (ErikSchierboom) [inline Dockerfile]: Why insall both curl _and_ wget? Are both needed?

Comment (ingydotnet) [reply Dockerfile]: Fixed in 997087b

**Comment by ingydotnet** [comment]
> @ingydotnet The TAP output looks off, it seems some of the tests indicate they're expecting `nil` rather than `false`?

Fixed in 5ae9c5d

**Review by ErikSchierboom — APPROVED** [review]
